electrumsv/cached_headers.py

```diff
@@ -1,82 +1,260 @@
+from __future__ import annotations
```

Comment:
I will approve the pull request but I wonder if this file would benefit from a multi-line comment at the very top giving an overview of the general concept here.

i.e. I suspect it could save new-comers to this code a fair bit of time to understand why all of this is being done and that there is both a standard `headers2` `bitcoinx.Headers` mmap file and a _lagging_ `headers2.chain_data` flat file representation. "Unprocessed" headers as you have called them are headers that are present in the `headers2` store but which are _yet to be written_ to the `headers2.chain_data` flat file representation (which involves overwriting the metadata and file hash at the beginning).

And one of the key advantages is that it allows an alternative `__init__` of the application `Headers` object which skips the (slow, iterative) reading in of all the headers every time the ESV wallet is loaded.

Additionally, am I right in saying that the only way to have "Unprocessed" headers (i.e. a state in which the `headers2.chain_data` is not yet synchronized with the latest headers from `headers2`) is if the wallet does not complete a graceful shutdown for whatever reason?

Worth mentioning all of these types of things in a general docstring?